casadm/cas_lib.c

```diff
@@ -1825,9 +1825,12 @@ int check_if_mounted(int cache_id, int core_id)
 	}
 
 	while ((mstruct = getmntent(mtab)) != NULL) {
+		mnt_fsname_len = strnlen(mstruct->mnt_fsname, sizeof(dev_buf));
+		error = strncmp_s(mstruct->mnt_fsname, mnt_fsname_len, dev_buf, dev_buf_len, &difference);
```

Comment:
Couple comments:
1. strncm_s  requires max buffer sizes for both strings rather than exact "strnlen" of the string. 
2. mstruct->mnt_fsname is provided by standard library call, so it would be wise to assume that max buffer length is PATH_MAX. 
3. It looks that now this code would treat two string if equal if mnt_fsname is a prefix of /dev/cas*. This is not correct.

I would replace these two lines with simple:
error = strncmp_s(mstruct->mnt_fsname, PATH_MAX, dev_buf, dev_buf_len, &difference);